### PR TITLE
feat: Complete forum channel implementation

### LINKED ISSUE #1635: Missing parameters and properties for ForumChannel Type
### Summary

Support for forum tags was added in the API

### What is the feature request for?

The core library

### The Problem

Discord finally released the tag properties for guilds. 
Which allows us to interact with forum tags through our bots.

The tag object:
https://discord.com/developers/docs/resources/channel#forum-tag-object

Missing properties found here:
https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel-jsonform-params

Especially since the forum feature itself was released to the public literally 15 minutes after I posted this feature request.

### The Ideal Solution

Implemented into the library for people to use.

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Dorukyum**
## Summary
Closes #1635.

## Information
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist
- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

File: discord/types/channel.py [hunk 1]
```diff
@@ -71,6 +71,14 @@ class TextChannel(_BaseGuildChannel, _TextChannelOptional):
     type: Literal[0]
 
 
+class ForumTag(TypedDict):
+    id: Snowflake
+    name: str
+    moderated: bool
+    emoji_id: Snowflake
+    emoji_name: str
```
Comment (Lulalaby) [inline discord/types/channel.py:79]: can be None
moderate might default to false

**Comment by Dorukyum** [comment]
I've added the `applied_tags` field to `create_thread` and `available_tags` attribute to `_TextChannel` as a list of integers since there's no `ForumTag` class as of now. `Thread.edit` must be completed and a `ForumTag` class should probably be created.
CC @BobDotCom

**Comment by Lulalaby** [comment]
ping pong conflicts and needs works
high prio

File: discord/http.py [hunk 2]
```diff
@@ -1149,6 +1154,7 @@ def start_forum_thread(
         auto_archive_duration: threads.ThreadArchiveDuration,
         rate_limit_per_user: int,
         invitable: bool = True,
+        applied_tags = SnowflakeList | None,
```
Comment (BobDotCom) [inline discord/http.py]: Shouldn't this have a default value?
```suggestion
        applied_tags = SnowflakeList | None = None,
```

File: discord/http.py [hunk 3]
```diff
@@ -1149,6 +1154,7 @@ def start_forum_thread(
         auto_archive_duration: threads.ThreadArchiveDuration,
         rate_limit_per_user: int,
         invitable: bool = True,
+        applied_tags: SnowflakeList | None,
```
Comment (BobDotCom) [inline discord/http.py:1157]: Shouldn't this have a default value?
```suggestion
        applied_tags: SnowflakeList | None = None,
```

**Comment by Lulalaby** [comment]
Top Prio, please finish

**Review by plun1331 — COMMENTED** [review]
available_tags should be under ForumChannel as opposed to _TextChannel probably

**Comment by Lulalaby** [comment]
What's missing

**Review by BobDotCom — DISMISSED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**